client/src/test/java/se/fortnox/reactivewizard/client/HttpClientTest.java

```diff
@@ -938,6 +939,68 @@ public void shouldExecutePreRequestHooks() throws URISyntaxException {
         server.shutdown();
     }
 
+    @Test
+    public void shouldHandleHttpsAgainstKnownHost() throws URISyntaxException {
+        HttpClientConfig httpClientConfig = new HttpClientConfig("https://sha512.badssl.com/");
+        Injector injector = TestInjector.create(binder -> {
+            binder.bind(ServerConfig.class).toInstance(new ServerConfig() {{
+                setEnabled(false);
+            }});
+            binder.bind(HttpClientConfig.class).toInstance(httpClientConfig);
+        });
+        RxClientProvider rxClientProvider = injector.getInstance(RxClientProvider.class);
+
+        rxClientProvider
+            .clientFor(new InetSocketAddress(httpClientConfig.getHost(), httpClientConfig.getPort()))
+            .createGet("/")
+            .flatMap(HttpClientResponse::discardContent)
+            .toBlocking()
+            .singleOrDefault(null);
+    }
+
+    @Test
+    public void shouldErrorOnUntrustedHost() throws URISyntaxException {
+        HttpClientConfig httpClientConfig = new HttpClientConfig("https://untrusted-root.badssl.com");
+        Injector injector = TestInjector.create(binder -> {
+            binder.bind(ServerConfig.class).toInstance(new ServerConfig() {{
+                setEnabled(false);
+            }});
+            binder.bind(HttpClientConfig.class).toInstance(httpClientConfig);
+        });
+        RxClientProvider rxClientProvider = injector.getInstance(RxClientProvider.class);
+
+        try {
+            rxClientProvider
+                .clientFor(new InetSocketAddress(httpClientConfig.getHost(), httpClientConfig.getPort()))
+                .createGet("/")
+                .flatMap(HttpClientResponse::discardContent)
+                .toBlocking()
+                .singleOrDefault(null);
```

Comment:
add a fail in the end, so that test won't go green even if the exception isn't thrown.